tfhe/src/core_crypto/entities/seeded_ggsw_ciphertext_list.rs

```diff
@@ -466,3 +467,77 @@ impl<Scalar: UnsignedInteger, C: ContainerMut<Element = Scalar>> ContiguousEntit
     where
         Self: 'this;
 }
+
+pub struct GgswCiphertextListConformanceParameters {
+    pub len: usize,
+    pub glwe_size: GlweSize,
+    pub polynomial_size: PolynomialSize,
+    pub decomp_base_log: DecompositionBaseLog,
+    pub decomp_level_count: DecompositionLevelCount,
+    pub ciphertext_modulus: CiphertextModulus<u64>,
+}
+
+impl TryFrom<&MultiBitBootstrapKeyConformanceParams> for GgswCiphertextListConformanceParameters {
+    type Error = ();
+
+    fn try_from(value: &MultiBitBootstrapKeyConformanceParams) -> Result<Self, ()> {
+        if value.input_lwe_dimension.0 % value.grouping_factor.0 != 0 {
+            return Err(());
+        }
+
+        let group_count = value.input_lwe_dimension.0 % value.grouping_factor.0;
+
+        Ok(Self {
+            len: group_count * value.grouping_factor.ggsw_per_multi_bit_element().0,
+            glwe_size: value.output_glwe_size,
+            polynomial_size: value.polynomial_size,
+            decomp_base_log: value.decomp_base_log,
+            decomp_level_count: value.decomp_level_count,
+            ciphertext_modulus: value.ciphertext_modulus,
+        })
+    }
+}
+
+impl From<&BootstrapKeyConformanceParams> for GgswCiphertextListConformanceParameters {
+    fn from(value: &BootstrapKeyConformanceParams) -> Self {
+        Self {
+            len: value.input_lwe_dimension.0,
+            glwe_size: value.output_glwe_size,
+            polynomial_size: value.polynomial_size,
+            decomp_base_log: value.decomp_base_log,
+            decomp_level_count: value.decomp_level_count,
+            ciphertext_modulus: value.ciphertext_modulus,
+        }
+    }
+}
+
+impl<C: Container<Element = u64>> ParameterSetConformant for SeededGgswCiphertextList<C> {
+    type ParameterSet = GgswCiphertextListConformanceParameters;
+
+    fn is_conformant(&self, parameter_set: &Self::ParameterSet) -> bool {
+        let Self {
+            data,
+            glwe_size,
+            polynomial_size,
+            decomp_base_log,
+            decomp_level_count,
+            compression_seed: _,
+            ciphertext_modulus,
+        } = self;
+
+        data.container_len()
+            == parameter_set.len
+                * ggsw_ciphertext_encryption_mask_sample_count(
+                    *glwe_size,
+                    *polynomial_size,
+                    *decomp_level_count,
+                )
+                .0
```

Comment:
that does not seem correct to me

Reply:
or maybe I'm missing something

Reply:
mask samples ? when we store only the bodies ?

Reply:
Fixed